.github/workflows/code_quality.yaml

```
@@ -43,7 +43,7 @@ jobs:
           python-version: '3.10'
       - name: Install dependencies
         run: |
-          pip install black
+          pip install black==22.12.0
```

```suggestion
          pip install "$(grep 'black==' setup.cfg)"
```

Not really pretty, but it's always a bit painful to have GH CI files in sync with the project dependencies

---

I prefer having libraries install/run the latest `black` version. If CI breaks, it's just a matter of installing the latest black locally and applying it on the main branch.

Releases are infrequent enough that make this possible IMO.

Pinning versions in setup.cfg for libraries is an anti-pattern, let's just let CI be the source of truth.

---

> Releases are infrequent enough that make this possible IMO.

Yes, we should expect breaking formatting changes only every year with black as per their policy.

> Pinning versions in setup.cfg for libraries is an anti-pattern

It is for direct dependencies, not necessarily for dev/test deps. Well here it's a bit different because we use an extra group for the dev deps :)